src/CommunityToolkit.Maui/Views/Popup/PopupPage.shared.cs

```diff
@@ -0,0 +1,185 @@
+using System.ComponentModel;
+using System.Globalization;
+using CommunityToolkit.Maui.Converters;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Maui.Controls.PlatformConfiguration;
+using Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;
+using Microsoft.Maui.Controls.Shapes;
+
+namespace CommunityToolkit.Maui.Views;
+
+sealed partial class PopupPage<T>(Popup<T> popup, IPopupOptions popupOptions) : PopupPage(popup, popupOptions)
+{
+	public PopupPage(View view, IPopupOptions popupOptions)
+		: this(view as Popup<T> ?? CreatePopupFromView<Popup<T>>(view), popupOptions)
+	{
+	}
+
+	public Task Close(PopupResult<T> result, CancellationToken token = default) => base.Close(result, token);
+}
+
+partial class PopupPage : ContentPage
+{
+	readonly Popup popup;
+	readonly IPopupOptions popupOptions;
+	readonly Command tapOutsideOfPopupCommand;
+	readonly WeakEventManager popupClosedEventManager = new();
+
+	public PopupPage(View view, IPopupOptions popupOptions)
+		: this(view as Popup ?? CreatePopupFromView<Popup>(view), popupOptions)
+	{
+		ArgumentNullException.ThrowIfNull(view);
+	}
+
+	public PopupPage(Popup popup, IPopupOptions popupOptions)
+	{
+		ArgumentNullException.ThrowIfNull(popup);
+		ArgumentNullException.ThrowIfNull(popupOptions);
+
+		this.popup = popup;
+		this.popupOptions = popupOptions;
+
+		// Only set the content if parent constructor hasn't set the content already; don't override content if it already exists
+		base.Content ??= new PopupPageLayout(popup, popupOptions);
+
+		tapOutsideOfPopupCommand = new Command(async () =>
+		{
+			popupOptions.OnTappingOutsideOfPopup?.Invoke();
+			await Close(new PopupResult(true));
+		}, () => popupOptions.CanBeDismissedByTappingOutsideOfPopup);
+
+		Content.GestureRecognizers.Add(new TapGestureRecognizer { Command = tapOutsideOfPopupCommand });
+
+		if (popupOptions is BindableObject bindablePopupOptions)
+		{
+			bindablePopupOptions.PropertyChanged += HandlePopupPropertyChanged;
+		}
+
+		this.SetBinding(BindingContextProperty, static (Popup x) => x.BindingContext, source: popup, mode: BindingMode.OneWay);
+		this.SetBinding(BackgroundColorProperty, static (IPopupOptions options) => options.PageOverlayColor, source: popupOptions, mode: BindingMode.OneWay);
+
+		Shell.SetPresentationMode(this, PresentationMode.ModalNotAnimated);
```

Comment:
Do we need to do something similar for `NavigationPage`?

Reply:
Do we need to do something similar for NavigationPage?

Nope! Already done.

We include `animated: false` when we call `Navigation.PushModalAsync`:

```cs
await Navigation.PushModalAsync(popup, animated: false)
```